### PR TITLE
server: move API Guidelines URL to config (#346)

### LINKED ISSUE #346: Move Restful API Guidelines base URL to config variable
While checking the existing rules I've found that Zalando Restful API Guidelines base URL is hard-coded in every rule. This leads to some inconsistencies, e.g. some of rules have `http` protocol and some `https`. 

Let's move base URL to application properties, so it will be consistent and we can change it any time.

### PR DESCRIPTION
**Pull request by innokenty**
Closes #346

### CODE REVIEW

**Comment by innokenty** [comment]
Any idea, why one test in cli-go started to fail? I don't see how it can be related: because there is no actual server running during tests, if I get it right.

**Comment by akryvko** [comment]
Actually we are running server, see in travis file -  ./gradlew bootRun > /dev/null &
And do integration testing against most recent server codebase
Probably output slightly changed:
#   `[[ "$output" =~ "	http://zalando.github.io/restful-api-guidelines/hyper-media/Hypermedia.html#must-do-not-use-link-headers-with-json-entities" ]]' failed

**Comment by innokenty** [comment]
Yes thanks, I see that the output has changed, but I have no idea why so far. I tried to find the config, with which the server is running during tests, but no success.

**Comment by netme** [comment]
Time to get rid of Bats 😃  ( #108 )

**Comment by netme** [comment]
@innokenty, as @akryvko mentioned it's running by Travis with default config, here are the exact lines: 
https://github.com/zalando-incubator/zally/blob/master/.travis.yml#L32-L36 

The failing test is doing `GET /supported-rules`, most probably the `ignoredRules` parameter of the default config is not taken into account:

https://travis-ci.org/zalando-incubator/zally/builds/257737109#L3214

**Comment by innokenty** [comment]
Thanks! No, the problem is different and related to changes in this PR. I started the server locally, requested the `/supported-rules` endpoint and I see that the base url from the config is not picked up. No idea why. Just all the rules have urls like this:
```
"url" : "/naming/Naming.html#should-reserved-javascript-keywords-should-be-avoided",
```

Whereas during server integration tests — it works!

**Comment by innokenty** [comment]
is `server/src/main/resources/rules-config.conf` even used?

**Comment by innokenty** [comment]
should be, `application.yml:33` says:
```
rules-config-path: "rules-config.conf"
```

**Comment by netme** [comment]
```
... and related to changes in this PR.
```
Definitely it is, the master is green :-)

Yeah, please check, I'm not that familiar with the config we are using. Can only say that it's working perfectly for the rules configuration, maybe it's initialized too late...

Why don't you move the base URL to `application.properties`?

**Comment by innokenty** [comment]
> the master is green
yeah, I already checked that with this funny PR: https://github.com/zalando-incubator/zally/pull/450

**Comment by innokenty** [comment]
I dunno, I can move. Just felt that it's kinda a rule config, but probably you're right

**Comment by netme** [comment]
There is a badge for that btw:

**Comment by innokenty** [comment]
Not sure what the issue was still. but I'm moving the property to `application.yml` and this way it works for all environments. Will force-push soon! 💪

**Comment by innokenty** [comment]
PLZ don't merge yet!

**Comment by innokenty** [comment]
And now it's ready. Also the changes got way more simple. Please review/merge!

File: server/src/test/java/de/zalando/zally/rule/RestSupportedRulesTest.java [hunk 1]
```diff
@@ -24,15 +25,22 @@
     @Autowired
     private List<Rule> implementedRules;
 
+    @Value("${zally.apiGuidelinesBaseUrl:something went wrong!}")
```
Comment (netme) [inline server/src/test/java/de/zalando/zally/rule/RestSupportedRulesTest.java]: Hehe, I like the default value :-)

**Comment by netme** [comment]
👍

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by maxim-tschumak** [comment]
:+1:

File: server/src/test/java/de/zalando/zally/rule/RestSupportedRulesTest.java (hunk 1 above)
Comment (innokenty) [inline server/src/test/java/de/zalando/zally/rule/RestSupportedRulesTest.java]: Yeah, I think it's quite a useful trick for the tests.

**Comment by innokenty** [comment]
@maxim-tschumak I think you need to actually click "approve" now, so that it could be merged.

**Comment by maxim-tschumak** [comment]
@innokenty No, there should be 2x :+1: after the last commit. @netme has to approve again :)

**Comment by innokenty** [comment]
Says the opposite on the top:
screen shot 2017-07-27 at 10 11 41

**Comment by innokenty** [comment]
kek

**Comment by maxim-tschumak** [comment]
Zappr is the one who doesn't allow to merge if there are not enough approvals (zappr approval = :+1: in the comments after the last change) - I know, approval definition is somehow overloaded here :)

**Comment by innokenty** [comment]
Ah, wow, so that's a feature!! I thought you're just thumbs-upping out of excitement 😆

**Comment by netme** [comment]
👍

**Pull request merged**